cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -588,8 +588,29 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 	}
 
 	if !p.ignoreDeploymentState && parametersHashErr == nil {
-		deploymentState, err := p.deploymentState(ctx, planned, deployment, currentParamsHash)
-		if err == nil {
+		deploymentState, stateErr := p.deploymentState(ctx, planned, deployment, currentParamsHash)
+		if stateErr == nil {
+			// As a heuristic, we also check the existence of all resource groups
+			// created by the deployment to validate the deployment state.
+			// This handles the scenario of resource group(s) being deleted outside of azd,
+			// which is quite common.
+			// This check adds ~100ms per resource group to the deployment time.
+			for _, res := range deploymentState.Resources {
```

Comment:
Should this be recursive?  In case `Resources` contains a resource type deployment which contains rg?

Reply:
Great question! Recursion isn't necessary here. The `outputResources` property from ARM already contains all resource(s) created from the current and nested deployments.